src/features/registry.js

```diff
@@ -67,7 +67,7 @@ systemJSPrototype.delete = function (id) {
   var load = registry[id];
   // in future we can support load.E case by failing load first
   // but that will require TLA callbacks to be implemented
-  if (!load || load.e !== null || load.E)
+  if (!load || load.E)
```

Comment:
As explained in https://github.com/systemjs/systemjs/issues/2286#issuecomment-748372786, I think we should allow deletion of modules that have been loaded, but not instantiated. Not sure if this has other negative consequences for other situations.